datafusion/substrait/src/logical_plan/consumer.rs

```diff
@@ -426,12 +426,12 @@ pub async fn from_substrait_rel(
                         new_name = format!("{}__temp__{}", name, i);
                         i += 1;
                     }
-                    names.insert(new_name.clone());
                     if new_name != name {
-                        exprs.push(x.as_ref().clone().alias(new_name.clone()));
+                        exprs.push(x.alias(new_name.clone()));
```

Comment:
because the code returned `Arc<Expr>` previously, but most places in `Expr` need either `Expr` or `Box<Expr>` the code ended up copying the expressions unnecessarily